butter.py

```diff
@@ -35,41 +35,59 @@ def __init__(self, *args, **kwargs):
         baseReader = BaseReader(self.settings)
 
-        butter_result = self.client.posts.all()
-        all_articles = []
-        if 'data' in butter_result:
-            posts = butter_result['data']
-            for post in posts:
-                if post['status'] == 'published':
-                    datestr = post['published'] if 'published' in post else post['created']
-                    date = parser.parse(datestr)
-                    title = post['title']
-                    content = post['body']
-                    author = post['author']['first_name']
-                    authorObject = baseReader.process_metadata('author', author)
-                    slug = post['slug']
-                    if post['categories']:
-                        category = post['categories'][0]['name']
-                    else:
-                        category = ''
-                    categoryObj = baseReader.process_metadata('category', category)
-
-                    metadata = {'title': title,
-                                'date': date,
-                                'category': categoryObj,
-                                'authors': [authorObject],
-                                'slug': slug}
-
-                    article = Article(content=content,
-                                      metadata=metadata,
-                                      settings=self.settings,
-                                      context=self.context)
-
-                    # # This seems like it cannot happen... but it does without fail.
-                    article.author = article.authors[0]
-                    all_articles.append(article)
+        butter_posts = []
+        page = 1
+        while True:
+            # Paginate through all pages
+            result = self.client.posts.all({'page_size': 10, 'page': page})
+            if 'data' in result:
+                butter_posts.extend(result['data'])
+
+            if 'meta' in result and 'next_page' in result['meta'] and result['meta']['next_page']:
+                page += 1
+            else:
+                break
+        # all_articles = []
+        counter = 0
+        for post in butter_posts:
+            if post['status'] == 'published':
+                counter += 1
+                logger.info('GET TO article: %s' % post['title'])
+                logger.info('counter: %s' % str(counter))
+                datestr = post['published'] if 'published' in post else post['created']
+                date = parser.parse(datestr)
+                title = post['title']
+                content = post['body']
+                author = post['author']['first_name']
+                authorObject = baseReader.process_metadata('author', author)
+                slug = post['slug'] if 'slug' in post else None
+                logger.info('--HAS slug: %s' % str(slug))
+                categoryObj = None
+                if post['categories']:
+                    category = post['categories'][0]['name']
```

Comment:
Can Pelican only support one category? Posts can have multiple categories.

Reply:
Yes, Pelican only supports 1 single category. To be able to support multiple categories, it needs plugins.